### DISCUSSION TITLE
Auto play cache file?

### DISCUSSION THREAD

**Discussion by JasonSleigh** [bluefireteam/audioplayers#757] [General]
I have a project where I would like to play a sound on the entrance to a page. I like the cache method as all my files are local. Is there a way to have the player play a sound when a user enters the page?

**Comment by erickzanardo** [comment]
AudioPlayers don't implement any UI resources, we focus on the audio part only.

But I imagine that there are a lot of ways of doing that, one that promptly comes to my mind is to call the play method on a `initState` of the State of the widget of your page.

**Reply by JasonSleigh** [reply]
Thanks,
I'll have a look at that sorry if this was the wrong place to ask.

**Reply by erickzanardo** [reply]
Don't worry! Discussions is the perfect place to ask things like that!

**Reply by luanpotter** [reply]
You can also play it on the screen transition method to make sure it's only played when you want and not when the widget tree gets rebuilt for some other reason.

**Comment by JasonSleigh** [comment]
Thanks Luan will remember that
